### PR TITLE
Initial commit to discharge throughput normalization

### PR DESCRIPTION
**Pull request by BruisVanVlijmen2020**

### CODE REVIEW

**Comment by patrickherring-TRI** [comment]
I think the failing test is due to this line: https://github.com/TRI-AMDD/beep/blob/73724cb76bbc97543f745253775c98967b98d318/beep/tests/test_features.py#L201

Probably needs to be changed to match the alteration in the method

File: beep/features/featurizer_helpers.py [hunk 1]
```diff
@@ -886,3 +887,57 @@ def get_fractional_quantity_remaining(
...
     summary_diag_cycle_type.columns = ["cycle_index", "fractional_metric"]
     return summary_diag_cycle_type
+
+
+def get_fractional_quantity_remaining_nx(
+    processed_cycler_run, metric="discharge_energy", diagnostic_cycle_type="rpt_0.2C"
+):
+    """
+    Similar to get_fractional_quantity_remaining()
+    Determine relative loss of <metric> in diagnostic_cycles of type <diagnostic_cycle_type> after 100 regular cycles
```
Comment (chirranjeevigopal-TRI) [inline beep/features/featurizer_helpers.py:897]: Can you also update the docstring here: need to remove "after 100 regular cycles"

**Comment by BruisVanVlijmen2020** [comment]
I changed updated that now. Pytests on test_features.py run without errors.
But still the end to end test throws the same error: FAILED beep/tests/test_end_to_end.py::EndToEndTest::test_console - json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

File: beep/features/featurizer_helpers.py [hunk 2]
```diff
@@ -886,3 +887,57 @@ def get_fractional_quantity_remaining(
...
+    Also returns value of 'x', the discharge throughput passed by the first diagnostic
+    and the value 'n' at each diagnostic
+
+    Args:
+        processed_cycler_run (beep.structure.ProcessedCyclerRun): information about cycler run
+        metric (str): column name to use for measuring degradation
+        diagnostic_cycle_type (str): the diagnostic cycle to use for computing the amount of degradation
+
+    Returns:
+        a dataframe with cycle_index and corresponding degradation relative to the first measured value
```
Comment (chirranjeevigopal-TRI) [inline beep/features/featurizer_helpers.py:907]: Update this to reflect what the method return (dataframe with cycle_index, fractional_metric, x, n)

**Comment by patrickherring-TRI** [comment]
@BruisVanVlijmen2020 The tests are passing on the PR (the only fail is due to a coverage decrease)

**Review by chirranjeevigopal-TRI — APPROVED** [review]
Thanks for the PR Bruis, and the changes. We can look into your tests on the local machine to see if its something about the test set-up, or some print statements throwing the error.

**Pull request merged**